core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -654,7 +666,7 @@ public CompletableFuture<Optional<Throwable>> releaseAcquiredRecords(String memb
                             }
                             // Successfully updated the state of the offset.
                             updatedStates.add(updateResult);
-                            localNextFetchOffset = Math.min(offsetState.getKey(), localNextFetchOffset);
+                            if (updateResult.state != RecordState.ARCHIVED) localNextFetchOffset = Math.min(offsetState.getKey(), localNextFetchOffset);
```

Comment:
The code style in the rest of this method suggests that the if test and the statement should be on separate lines. There's another instance of this too.

Reply:
Missed it, updated the code